File: lab3/java/StorageProducer.java
public class StorageProducer implements Runnable {
	private final int itemAmount;
	private final StorageManager manager;
	
	public StorageProducer (StorageManager manager, int itemAmount) {
		this.manager = manager;
		this.itemAmount = itemAmount;
		
		new Thread(this).start();
	}
	
	@Override
	public void run () {
		for (int i = 1; i <= itemAmount; i++) {
			try {
				manager.getAddItemPermit();
				manager.getAccessPermit();
				
				manager.addItem("item " + i + " [ " + this + " ]");
				System.out.println("Added item " + i + " [ " + this + " ]");

				Thread.sleep(1000);
				manager.dropAccessPermit();
				manager.addNewTakeItemPermit();
			} catch (InterruptedException e) {
				e.printStackTrace();
			}
		}
	}
}

File: lab3/cs/StorageProducer.cs
namespace Laba3_Semaphores_CSharp.classes;

public class StorageProducer
{
  private int itemsAmount;
  private StorageManager manager;
  private int idx;
	
  public StorageProducer (StorageManager manager, int itemsAmount, int idx) {
    this.manager = manager;
    this.itemsAmount = itemsAmount;
    this.idx = idx;
		
    new Thread(Start).Start();
  }

  public void Start()
  {
    for (int i = 0; i < itemsAmount; i++)
    {
      manager.getAddItemPermit();
      manager.getAccessPermit();

      manager.addItem($"item {i} [ {this.ToString().Split('.')[^1]}-{idx} ]");
      Console.WriteLine($"Added item {i} [ {this.ToString().Split('.')[^1]}-{idx} ]");

      manager.dropAccessPermit();
      manager.addNewTakeItemPermit();
    }
  }
}